app/validators/empty_profile_field_names_validator.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class EmptyProfileFieldNamesValidator < ActiveModel::Validator
+  def validate(account)
+    return if account.fields.empty?
+
+    account.errors.add(:fields, I18n.t('edit_profile.errors.fields_with_values_missing_names')) if fields_with_values_missing_names?(account)
+  end
+
+  private
+
+  def fields_with_values_missing_names?(account)
+    account.fields.each do |field|
+      return true if field.name.blank? && field.value.present?
+    end
+    false
```

Comment:
I think this is better written as:
```suggestion
    account.fields.any? { |field| field.name.blank? && field.value.present? }
```

Reply:
Makes sense, thanks for the suggestion! I have updated it.